core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -251,6 +251,14 @@ public static ElasticsearchParseException newConfigurationException(String proce
         return processors;
     }
 
+    public static TemplateService.Template compileTemplate(String processorType, String processorTag, String propertyName, String propertyValue, TemplateService templateService) {
+        try {
+            return templateService.compile(propertyValue);
+        } catch (Exception e) {
+            throw ConfigurationUtils.newConfigurationException(processorType, processorTag, propertyName, e.getMessage());
+        }
```

Comment:
Please don't lose the original exception.  It's already difficult enough to debug script exceptions without them being swallowed.